packages/container/tests/container_test.js

```diff
@@ -615,3 +616,44 @@ QUnit.skip('#factoryFor does not add properties to the object being instantiated
   // not via registry/container shenanigans
   assert.deepEqual(Object.keys(instance), []);
 });
+
+if (isFeatureEnabled('ember-module-unification')) {
+  QUnit.module('Container module unification');
+
+  QUnit.test('The container can pass a source to factoryFor', function() {
+    let PrivateComponent = factory();
+    let src = 'template:routes/application';
+    let registry = new Registry();
+    registry.resolve = function(fullName, { source }) {
+      if (fullName === 'component:my-input' && src === source) {
+        return PrivateComponent;
+      }
+    };
+
+    let container = registry.container();
+
+    strictEqual(container.factoryFor('component:my-input', { source: src }).class, PrivateComponent, 'The correct factory was provided');
```

Comment:
Tests are passed `assert` as an argument and should use that over the global assertion methods. For example:

```js
QUnit.test('The container can pass a source to factoryFor', function(assert) {
  /* ... */
  assert.strictEqual(
    container.factoryFor('component:my-input', {source: src}).class, PrivateComponent,
    'The correct factory was provided'
  );
  /* ... */
```

In fact re-using the string `'component:my-input'` instead of re-typing it would clean things up as well, as would using `source` instead of `src` as the source string variable name (then this can be `{source}`).